junit5-extension/src/main/java/io/kroxylicious/testing/kafka/junit5ext/KafkaClusterExtension.java

```diff
@@ -970,13 +971,28 @@ private static Admin getAdmin(String description,
                     LOGGER.log(TRACE, "test {0}: decl {1}: Creating KafkaConsumer<>",
                             extensionContext.getUniqueId(),
                             sourceElement);
-                    return new Closeable<>(sourceElement, cluster.getClusterId(), new KafkaConsumer<>(cluster.getKafkaClientConfiguration(),
+                    return new Closeable<>(sourceElement, cluster.getClusterId(), new KafkaConsumer<>(buildConfig(sourceElement, cluster),
                             keySerializer, valueSerializer));
                 },
                         (Class<Closeable<KafkaConsumer<?, ?>>>) (Class) Closeable.class)
                 .get();
     }
 
+    private static Map<String, Object> buildConfig(AnnotatedElement sourceElement, KafkaCluster cluster) {
```

Comment:
Feels a bit odd to need this here and in `impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java`